Library/Homebrew/cmd/audit.rb

```diff
@@ -548,12 +548,14 @@ def audit_github_repository
   end
 
   def audit_specs
-    if head_only?(formula) && formula.tap.to_s.downcase !~ /-head-only$/
-      problem "Head-only (no stable download)"
-    end
+    if formula.tap
```

Comment:
If we decide to go with `formula.tap.repo` instead of `formula.tap.to_s`, that's probably what the code in question should look like:

```ruby
if head_only?(formula) && formula.tap && formula.tap.repo !~ /(^|-)head-only$/i
  problem "Head-only (no stable download)"
end
```

(Note that this is semantically different from what @xu-cheng suggested, prefers a case-insensitive match instead of using `downcase`, and accepts both `head-only` and `<prefix>-head-only` in the repo part, but not `<prefix>head-only`.)